File: Python3/Medium/Accepted/subdomain-visit-count_8.py
class Solution:
    def subdomainVisits(self, cpdomains: List[str]) -> List[str]:
        
        domain_dic = {}
        for cpdomain in cpdomains:
            count, domain = cpdomain.split(' ')
            count = int(count)
            domain_list = domain.split('.')
            for i in range(len(domain_list)):
                domain = '.'.join(domain_list[i:])
                if domain in domain_dic:
                    domain_dic[domain] += count
                else:
                    domain_dic[domain] = count
        return [str(v) + ' ' + k for k, v in domain_dic.items()]

File: Java/Medium/Accepted/subdomain-visit-count_8.java
class Solution {
    public List<String> subdomainVisits(String[] cpdomains) {
        
        List<String> result = new ArrayList<>();
        HashMap<String, Integer> map = new HashMap<>();
        for (String cpdomain : cpdomains) {
            String[] cpInfo = cpdomain.split(" ");
            int count = Integer.parseInt(cpInfo[0]);
            String domain = cpInfo[1];
            map.put(domain, map.getOrDefault(domain, 0) + count);
            for (int i = 0; i < domain.length(); i++) {
                if (domain.charAt(i) == '.') {
                    String subdomain = domain.substring(i + 1);
                    map.put(subdomain, map.getOrDefault(subdomain, 0) + count);
                }
            }
        }
        for (String key : map.keySet()) {
            result.add(map.get(key) + " " + key);
        }
        return result;
    }
}